pkgs/os-specific/linux/rasdaemon/default.nix

```
@@ -0,0 +1,102 @@
+{ stdenv, lib, fetchurl, automake, autoreconfHook, coreutils, dmidecode
+, glibcLocales, hwdata, kmod, libtool, makeWrapper, sqlite, perl, doCheck ? true
+}:
+stdenv.mkDerivation rec {
+  name = "rasdaemon";
+  version = "0.6.4";
+  src = fetchurl {
+    url =
+      "http://www.infradead.org/~mchehab/rasdaemon/rasdaemon-${version}.tar.bz2";
+    sha256 = "0p107x5nl5fm6n46sv45fk596z63j33j13aw076rs09m6bsgfwm6";
+  };
+
+  nativeBuildInputs = [ automake libtool makeWrapper ];
+  buildInputs = [
+    coreutils
+    dmidecode
+    glibcLocales
+    hwdata
+    kmod
+    sqlite
+    (perl.withPackages (ps: with ps; [ DBI DBDSQLite ]))
+  ];
+
+  dontAddDisableDepTrack = true;
+
+  configureFlags = [
+    "--sysconfdir=/etc"
+    "--localstatedir=/var"
+    # in debian and arch linux
+    "--enable-abrt-report"
+    "--enable-aer"
+    "--enable-arm"
+    "--enable-extlog"
+    "--enable-mce"
+    "--enable-sqlite3"
+    # in arch linux
+    "--enable-devlink"
+    "--enable-diskerror"
+    "--enable-hisi-ns-decode"
+    "--enable-non-standard"
+  ];
+
+  # The Makefile attempts to create the following directories:
+  # /var/lib/rasdaemon
+  #   location of the RAS event log generated by rasdaemon -r
+  # /etc/ras/dimm_labels.d
+  #   location of the DIMM labels generated by ras-mc-ctl
+  # We can not create these directories at install time
+  # since the nix builder does not have permissions.
+  # Since these programs need permissions to these directories
+  # to run properly, we wrap the executables to create
+  # these directories on startup. (see postFixup)
+  postConfigure = ''
+    sed -e '/$(install_sh) -d "$(DESTDIR)\/var\/lib\/rasdaemon"/d' \
+    -e '/$(install_sh) -d "$(DESTDIR)\/etc\/ras\/dimm_labels.d"/d' \
+    -i Makefile
+  '';
+
+  inherit doCheck;
+
+  installFlags = [
+    "sysconfdir=${placeholder "out"}/etc"
+    "localstatedir=${placeholder "out"}/var"
+  ];
+
+  postInstall = ''
+    mkdir -p $out/lib/systemd/system
+    install -D -p -m 0644 misc/rasdaemon.service \
+      $out/lib/systemd/system/rasdaemon.service
+    install -D -p -m 0644 misc/ras-mc-ctl.service \
+      $out/lib/systemd/system/ras-mc-ctl.service
+  '';
+
+  postFixup = ''
+    # fix dmidecode and modprobe paths
```

```suggestion
    # Fix dmidecode and modprobe paths
```

Nitpick: Please start all your comment with a capital letter, just looks nicer :)